Docs for Analytics Rules plugin [PAB-4742] (#3395)
SoftwareAG/c8y-docs: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md

        An input block specifies a device or a range of devices, while an output block specifies a device, a trigger device or an asset. For template parameters, the same template parameter and thus value can be used for both input and output blocks. If a template parameter is set to refer to a range of devices, then using it in an output block will be treated as the trigger device. Typically, a single template parameter would be used for all input and output blocks, and may be a single device or a range of devices, in which case the block output goes to the device within the range that triggered a model evaluation \(so a model calculating an average of a measurement and outputting to a measurement would generate a new measurement for each device independently\). Even if a different template parameter whose value refers to a different range was used, the model output would only be sent to the device that triggered a model's evaluation.

    -   **Source or Destination Type**.

>         You can select one or more of the following "Source or Destination" types from **Restrict to** dropdown (multi-selectable): "Device", "Groups", "Assets", or "Other". This selection restricts the template parameter to the chosen types. This restriction is mandatory when "From Context" is selected from the **Value Selection** dropdown.

Reviewer: ```suggestion
        You can select one or more of the following source or destination types from the **Restrict to** dropdown: "Device", "Groups", "Assets", or "Other". This restricts the template parameter to the selected types. The restriction is mandatory when "From context" is selected from the **Value selection** dropdown.
```

Author: updated

        ![Template parameter with From Context](/images/streaming-analytics/analytics-builder/from-context-restrict-to.png)

>     -   **Value Selection**. Select one of the following options from the dropdown:

Reviewer: ```suggestion
    -   **Value selection**. Select one of the following options from the dropdown:
```

>         -   **Required** (default). Instance deployment requires a value for this template parameter.

Reviewer: ```suggestion
        -   Required - Default. Instance deployment requires a value for this template parameter.
```

>         -   **Optional**. The template parameter can remain blank or can be set later by the instance maintainer. When you select this option, it is not possible to specify a default value.

Reviewer: ```suggestion
        -   Optional - The template parameter can remain blank or can be set later by the instance maintainer. When you select this option, it is not possible to specify a default value.
```

Author: done

>         -   **From Context**. Applicable only for "Source or Destination" type. The input will be automatically chosen based on the context where the instance is created. Example: Device context or Groups context.

Reviewer: ```suggestion
        -   From Context - Applicable only for "Source or destination" type. The input will be automatically selected based on the context where the instance is created. Example: Device context or group context.
```

Author: updated

{{< c8y-admon-info>}}

> Only one template parameter per model can have "From Context" value selection.

Reviewer: ```suggestion
Only one template parameter per model can have a "From context" value selection.
```

Author: @BeateRixen **From Context** is the value we choose from dropdown?  Shouldn’t we keep it the same in the documentation for consistency?

Author: ![highlight](https://github.com/user-attachments/assets/7e323f9d-94ce-407a-827e-1f5c1c1717e5)
@BeateRixen Please check this. Value Selection and From Context are labels and value from dropdown respectively.

Reviewer: You are right, the documentation should be consistent with the UI. But according to the UI style guide, we use sentence capitalization: https://styleguide.cumulocity.com/apps/codex/?_gl=1*cbor49*_gcl_au*MTA0NjcwNDQwNS4xNzQ3MTMwNTc5LjEzNjM0NzA4ODQuMTc0OTU2Mjc3My4xNzQ5NTYyNzc3*_ga*MjAxMjE2NDQ1LjE3MzkyODY0NjA.*_ga_B9NM1M9Q48*czE3NTA3NTc2NTMkbzEyNSRnMSR0MTc1MDc1NzY1MyRqNjAkbDAkaDA.#/ui-guidelines/foundations/writing/grammar-and-conventions/overview#capitalization

So the UI spelling is actually incorrect. I leave it up to you/your team, how to handle this.

Reviewer: If you decide to leave as is, then I can modify my suggestions accordingly. Please also be aware that values are formatted in quotes.

Author: We will proceed with the current change for now and raise this topic in a broader discussion about updating the labels within the streaming analytics application.

{{< /c8y-admon-info>}}
    

>     -   **Default Value**. You can only specify a default value when the **Value Selection** option is set to **Required**.

Reviewer: ```suggestion
    -   **Default value**. You can only specify a default value when the **Value selection** option is set to "Required".
```
